### ISSUE TITLE
Add how to work with Windows Auth using Visual Studio Code

### ISSUE THREAD

**Issue by DanJ210** [aspnet/Docs#9357]
**Help us make content visible**

Did not find explanation on how to use Windows Auth with Visual Studio Code.

**Describe the new topic**

Visual studio code is huge and rapidly becoming a preferred IDE. I can use Windows Auth and debug when using regular Visual Studio using the docs but there's nothing about how to do it with Visual Studio Code. So I cannot debug an app needing Windows Auth with Visual Studio code and I feel like this is important to at least go over in those docs. They need the addition of this top notch IDE by Microsoft.

**Comment by guardrex** [comment]
Hello @DanJ210 ... There's not much to say about VSC with Windows Auth. It's mostly configured in IIS. The IIS Options default to a config that makes Windows Auth work OOB as far as the app is concerned (when using `CreateDefaultBuilder`). See: https://docs.microsoft.com/aspnet/core/host-and-deploy/iis/index#iis-options

If you have something specific that you think we need to add to https://docs.microsoft.com/aspnet/core/security/authentication/windowsauth that would be good for VSC users to see, please let us know.

**Comment by DanJ210** [comment]
Hey @guardrex I was mainly talking about the fact that I can run and debug a .Net Core app with Windows Auth in Visual Studio and I guess that's because Visual Studio runs an instance of IIS Express at the same time.

If I try to run the same app in VS Code, I can't because the Windows Auth identity objects don't get populated and the `[Authorize]` attributes won't work and I'm guessing that's because there's no IIS instance passing them to the app, I'm only guessing.

Perhaps this is part of my lack of knowledge while I learn in how to use these tools but with the above post I was mainly referring to how to run and debug using VS Code when apps have Windows Auth security going on and controllers are marked with the `[Authorize]` attributes. Is there a way to do this with VS Code if the app seems to rely on IIS? That's the kind of doc I would look over.

**Comment by guardrex** [comment]
Yes, that's correct. `iisexpress.exe` is spawned for VS debug runs when the **IIS Express** launch profile is used. Looking under the covers here with one of my test apps, the command that spawned it was ...

```console
"C:\Program Files\IIS Express\iisexpress.exe"  /config:"C:\_apps\testFDD\.vs\config\applicationhost.config" /site:"testFDD" /apppool:"testFDD AppPool"
```

VSC just runs the app with *dotnet.exe* ... here's what Process Explorer told me about `dotnet.exe` on a VSC debug run ...

```console
"C:\Program Files\dotnet\dotnet.exe" C:\_apps\testFDD\bin\Debug\netcoreapp2.2\testFDD.dll
```

Try running your app with IIS Express (*iisexpress.exe*) at a command prompt. If you can get it running that way from a command prompt, then you should be able to create a VSC task to run that command. ........... in theory anyway. :pray: Next assuming that goes well, see if you get the IIS Express system tray ... and then if so, see if you can use that to stop the IIS Express process to stop the app.

Let me know how it turns out. If something along those lines works and isn't too hacky, then perhaps @Rick-Anderson will decide that we should drop a line into this topic to cover it. Also, we don't cover the use of *iisexpress.exe* **_anywhere_** in the topics. It's an open question if we should attempt to resolve coverage of using IIS Express from the CLI and how we would cover it if the decision is made to do so.
